static/app/components/tabs/tabPanels.tsx

```diff
@@ -0,0 +1,93 @@
+import {useContext, useRef} from 'react';
+import styled from '@emotion/styled';
+import {AriaTabPanelProps, useTabPanel} from '@react-aria/tabs';
+import {useCollection} from '@react-stately/collections';
+import {ListCollection} from '@react-stately/list';
+import {TabListState} from '@react-stately/tabs';
+import {CollectionBase, Node, Orientation} from '@react-types/shared';
+
+import space from 'sentry/styles/space';
+
+import {TabsContext} from './index';
+
+interface TabPanelsProps<T> extends AriaTabPanelProps, CollectionBase<T> {
+  className?: string;
+}
+
+/**
+ * To be used as a direct child of the <Tabs /> component. See example usage
+ * in tabs.stories.js
+ */
+export function TabPanels<T>(props: TabPanelsProps<T>) {
+  const {
+    rootProps: {orientation, items},
+    tabListState,
+  } = useContext(TabsContext);
+
+  // Parse child tab panels from props and identify the selected panel
+  const factory = nodes => new ListCollection<Node<object>>(nodes);
+  const collection = useCollection({items, ...props}, factory);
+  const selectedPanel = tabListState
+    ? collection.getItem(tabListState.selectedKey)
+    : null;
+
+  if (!tabListState) {
+    return null;
+  }
+
+  return (
+    <TabPanel
+      {...props}
+      state={tabListState}
+      orientation={orientation}
+      key={tabListState?.selectedKey}
+    >
+      {selectedPanel && selectedPanel.props.children}
+    </TabPanel>
+  );
+}
+
+interface TabPanelProps<T> extends AriaTabPanelProps {
+  orientation: Orientation;
+  state: TabListState<T>;
+  children?: React.ReactNode;
+  className?: string;
+}
+
+function TabPanel<T>({state, orientation, className, ...props}: TabPanelProps<T>) {
+  const ref = useRef<HTMLDivElement>(null);
+  const {tabPanelProps} = useTabPanel(props, state, ref);
```

Comment:
do we need to pass `children` down here too?

Reply:
apparently not, detructured `children` in the props list in https://github.com/getsentry/sentry/pull/38886/commits/37f932bd776d2d052f71545b0d9cd4e4e55f2f02